tools/InstallationTests/AutoTestLogin.ps1

```diff
@@ -0,0 +1,490 @@
+
+Issues:
+1) During the very first log in using certificate, an error pops up "Login-AzureRmAccount : No certificate was found in the certificate store with thumbprint System.Object[]"
+This is a transient error, but eventually the login is successful as it finds the certificate in Currentuser store
+So after the first try, this error no longer shows up.
+Will open an issue to investigate where we try both the cert store simultenously rather we should try Currentuser and if not found then try LocalMachine.
+#>
+
+$global:gPsAutoTestADAppId = 'b8a1058e-25e8-4b08-b40b-d8d871dda591'
+$global:gPsAutoTestSubscriptionName = 'Node CLI Test'
+$global:gPsAutoTestSubscriptionId = '2c224e7e-3ef5-431d-a57b-e71f4662e3a6'
+
+$global:gTenantId = '72f988bf-86f1-41af-91ab-2d7cd011db47'
+$global:gLocalCertSubjectName = 'CN=PsAutoTestCert, OU=MicrosoftAzurePsTeam'
+$global:gPfxLocalFileName = "PsAutoTestCert.pfx"
+$global:gpubSettingLocalFileName = "NodeCLITest.publishsettings"
+
+$global:gVaultName = 'KV-PsSdkAuto'
+$global:gPsAutoResGrpName = 'AzurePsSdkAuto'
+
+$global:gLocalCertStore = 'Cert:\CurrentUser\My'
+$global:gCertPwd = ''
+$global:gLoggedInCtx = $null
+$global:localPfxDirPath = [Environment]::GetEnvironmentVariable("TEMP")
+
+#KV KeyVaules
+$global:kvSecKey_PsAutoTestCertNameKey = 'PsAutoTestCertName'
+$global:gPsAutoTestADAppUsingSPNKey = 'PsAutoTestADAppUsingSPN'
+$global:gkvSecKey_PubSettingFileNameKey = 'NodeCliTestPubSetFile'
+$global:gKVKey_ADAppIdKey = "PsAutoTestAppUsingCertAppId"
+
+
+Function Check-LoggedInContext()
+{
+    if($gLoggedInCtx -eq $null)
+    {
+        Write-Error "'$global:gPsAutoTestSubscriptionName' subcription does not exist in the list of available subscriptions. Make sure to have it to run the tests"
+        Exit
+    }
+}
+
+Function Get-AutomationTestCertificate()
+{
+    [OutputType([System.Security.Cryptography.X509Certificates.X509Certificate])]
+    Param ($cert)
+
+    if($gLoggedInCtx -ne $null)
+    {
+        #Download Certificate from KeyVault
+        $downloadedCertPath = Download-TestCertificateFromKeyVault
+
+        #Install Certificate locally
+        Install-TestCertificateOnMachine $downloadedCertPath
+
+        #Verify it was installed locally
+        $cert = Get-LocalCertificate
+        if($cert -eq $null)
+        {
+            throw [System.ApplicationException] "Unable to retrieve Automation Test Certificate '$gLocalCertSubjectName'"
+        }
+    }
+    return $cert
+}
+
+Function Get-LocalCertificate()
+{
+    #TODO: Handle case whgere we get multiple test certificates with exactly the same subject
+    $cert = Get-ChildItem $gLocalCertStore | Where-Object {$_.Subject -eq $Global:gLocalCertSubjectName}
+    if($cert -eq $null)
+    {
+        Log-Info "Trying to find certificate in LocalMachine"
+        $cert = Get-ChildItem 'Cert:\LocalMachine\My' | Where-Object {$_.Subject -eq $Global:gLocalCertSubjectName}
+    }
+
+    return $cert
+}
+
+Function Get-ADAppForAutoTest()
```

Comment:
Also, wondering about the name, this should be something like Ensure, rather than get.  It should return the ADApp, so that you don't have to retrieve it again later